recipes/certificate.rb

```diff
@@ -25,12 +25,16 @@
     cert_path node['abiquo']['certificate']['file']
     key_path  node['abiquo']['certificate']['key_file']
     not_if { ::File.file? "/etc/pki/abiquo/#{node['abiquo']['certificate']['common_name']}.crt" }
-    notifies :restart, 'service[apache2]'
+    only_if { node['abiquo']['certificate']['source'] == 'self-signed' }
+    only_if { node['abiquo']['profile'] != 'websockify' }
+    notifies :restart, 'service[apache2]' unless node['abiquo']['profile'] == 'websockify'
+    notifies :restart, 'service[websockify]' if node['abiquo']['profile'] == 'websockify'
```

Comment:
Hmmm there is the previous guard `only_if { node['abiquo']['profile'] != 'websockify' }`. Does this take effect?